examples/all-clusters-app/esp32/Makefile

```diff
@@ -26,7 +26,7 @@ EXTRA_COMPONENT_DIRS += $(PROJECT_PATH)/third_party/connectedhomeip/config/esp32
                         $(PROJECT_PATH)/../../common/QRCode \
                         $(PROJECT_PATH)/../../common/screen-framework \
 
-CXXFLAGS += -std=c++11 -Os -DLWIP_IPV6_SCOPES=0
+CXXFLAGS += -std=c++14 -Os -DLWIP_IPV6_SCOPES=0
```

Comment:
@bzbarsky-apple  Do you want to make similar change in other ESP32 examples? `examples/temperature-measurement-app/esp32/`

Reply:
Yep, good idea.  Done.